src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        const compareResult = compareFunction
+          ? compareFunction(this[i], this[j]) > 0
+          : String(this[i]) > String(this[j]);
```

Comment:
The ternary operation inside the loop is inefficient because it gets evaluated for every comparison even when a compare function is provided. Consider evaluating whether `compareFunction` is provided once before the loop starts, and then use the appropriate comparison logic within the loop.